Firestore/Example/Tests/Integration/API/FIRWriteBatchTests.mm

```diff
@@ -323,4 +324,53 @@ - (void)testUpdateNestedFields {
   [self awaitExpectations];
 }
 
+// Returns how much memory the test application is currently using, in megabytes (fractional part is
+// truncated), or -1 if the OS call fails.
+// TODO(varconst): move the helper function and the test into a new test target for performance
+// testing.
+long long getCurrentMemoryUsedInMb() {
```

Comment:
Note: there is also something like [performance tests](https://developer.apple.com/documentation/xctest/xctestcase/1496290-measureblock?language=objc) in `XCTest`. I originally had that here as well, but eventually dropped it for this case at least. It seems like a very limited and frankly a clunky feature. It's only possible to measure execution speed, the test is run 10 times which I think is impossible to configure, and to top it all, expected time can only be set in the UI (it is then stored in a file similar to a scheme).